code/lib/cli/src/doctor/getIncompatibleStorybookPackages.test.ts

```diff
@@ -0,0 +1,158 @@
+import { describe, it, expect, vi } from 'vitest';
+import type { AnalysedPackage } from './getIncompatibleStorybookPackages';
+import {
+  getIncompatibleStorybookPackages,
+  getIncompatiblePackagesSummary,
+  checkPackageCompatibility,
+} from './getIncompatibleStorybookPackages';
+import type { JsPackageManager } from '@storybook/core-common';
+
+import * as doctorUtils from './utils';
+
+vi.mock('chalk', () => {
+  return {
+    default: {
+      yellow: (str: string) => str,
+      cyan: (str: string) => str,
+      bold: (str: string) => str,
+    },
+  };
+});
+
+vi.mock('./utils', () => ({
+  getPackageJsonPath: vi.fn(() => Promise.resolve('package.json')),
+  getPackageJsonOfDependency: vi.fn(() => Promise.resolve({})),
+  PackageJsonNotFoundError: Error,
+}));
+
+const packageManagerMock = {
+  getAllDependencies: () =>
+    Promise.resolve({
+      '@storybook/addon-essentials': '7.0.0',
+    }),
+  latestVersion: vi.fn(() => Promise.resolve('8.0.0')),
+} as Partial<JsPackageManager>;
+
+describe('checkPackageCompatibility', () => {
+  it('returns that an package is incompatible', async () => {
+    const packageName = 'my-storybook-package';
+    vi.mocked(doctorUtils.getPackageJsonOfDependency).mockResolvedValueOnce({
+      name: packageName,
+      version: '1.0.0',
+      dependencies: {
+        '@storybook/core-common': '7.0.0',
+      },
+    } as any);
+    const result = await checkPackageCompatibility(packageName, {
+      currentStorybookVersion: '8.0.0',
+      packageManager: packageManagerMock as JsPackageManager,
```

Comment:
```suggestion
      packageManager: packageManagerMock,
```

Can we make it accept the partial?